lib/core/src/Cardano/Wallet/DB/Store/Transactions/Model.hs

```diff
@@ -276,3 +283,54 @@ mkTxHistory txs = TxHistoryF $ fold $ do
     tx <- txs
     let relation = mkTxRelation tx
     pure $ Map.singleton (TxId $ tx ^. #txId) relation
+
+type TxOutKey = (TxId, Word32)
+
+decorateWithTxOuts :: TxHistoryF 'Without -> TxHistoryF 'With
+decorateWithTxOuts (TxHistoryF w) =
+  let txouts :: Map TxOutKey (TxOut, [TxOutToken]) = Map.fromList $ do
+        TxRelationF {..} <- toList w
+        [(txOutputTxId &&& txOutputIndex $ txout, x) | x@(txout,_ ) <- outs]
+   in TxHistoryF $ fmap (solveTxOut txouts) w
```

Comment:
```suggestion
    let txouts :: Map TxOutKey (TxOut, [TxOutToken])
        txouts = Map.fromList $ do
            TxRelationF{..} <- toList w
            [(txOutputTxId &&& txOutputIndex $ txout, x) | x@(txout,_ ) <- outs]
    in  TxHistoryF $ fmap (solveTxOut txouts) w
```

(Style

1. Indentation multiple of 4 spaces (with exception for `where` clause – 2 spaces)
2. Repeating the name `txouts` for type signature and definition seems worth the clarity.
3. I prefer to have no space between record constructor `TxRelationF` and curly braces `{}`; in this way, I can visually distinguish function arguments by spaces.

)

Reply:
done, with a bit of doubts